c++/src/kj/async.c++

```diff
@@ -853,14 +920,52 @@ FiberBase::FiberBase(size_t stackSizeParam, _::ExceptionOrValue& result)
 #endif
 }
 
-FiberBase::~FiberBase() noexcept(false) {
+FiberStack::~FiberStack() noexcept(false) {
 #if _WIN32 || __CYGWIN__
   DeleteFiber(osFiber);
 #elif !__BIONIC__
   Impl::free(impl, stackSize);
 #endif
 }
 
+void FiberStack::initialize(FiberBase& fiber) {
+  KJ_ASSERT(this->fiber == nullptr);
```

Comment:
Nitpick: Use `KJ_REQUIRE` to check preconditions (failure = there is a bug in the caller). Use `KJ_ASSERT` is to test assumptions (failure = there's a bug in this code).

(In practice, they do exactly the same thing, but I like to use one or the other for self-documentation... admittedly I might eventually give up on this since even I mix them up fairly regularly.)